Commit recommendations meta page (#2604)
privacyguides/privacyguides.org: docs/meta/commit-messages.md

---
title: Commit Messages
---

For our commit messages we follow the style provided by [Conventional Commits](https://conventionalcommits.org). Not all of those suggestions are appropriate for Privacy Guides, so the main ones we use are:

## Commit message with correction

We use `fix` for simple things like spelling mistakes or site related bugs. These things will usually have the `correction` or `bug` label on GitHub.

```text
fix: Correct spelling on XYZ page (#0000)
```

## Update to site

> This example is for a software removal (but could be used for an addition), you may elaborate why it was removed in the commit paragraph below. It can also be used for any new pages that were added.

Reviewer: ```suggestion
This example is for a removal of an item (but could also be used for an addition); you may elaborate why it was removed in the commit paragraph below. It can also be used for the addition of any new pages.
```
This change might not be needed, but I think that "software" should be removed from the sentence to make this commit recommendation a more general one. After all, there was recent discussion on the forum about removing Nitrokey.

```text
update: Remove foobar (#0000)

Foobar was removed due to it having numerious security issues and being unmaintained.
```

## Update to specific item

This example could be used for an item already on the site, but includes a minor update to the description.

```text
foobar: Add mention of security audit (#0000)
```

## Feature/enhancement

> For new features or enhancements to the site, e.g. where we've used the `enhancements` label it may be appropriate to signify these with:

Reviewer: ```suggestion
For new features or enhancements to the site, e.g. things that have the `enhancements` label on GitHub, it may be appropriate to signify these with:
```
This changes the wording so that it is consistent with the "Commit message with correct" tip.
